### PR TITLE
Finish omero model

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/BioImageTools/ome-zarr-models-py/issues/34.

Note that in order to parse the example given in the spec and not lose the extra bits of metadata, I've enabled allowing extra fields in this PR (which I think we decided we should add at some point anyway?)

### CODE REVIEW

File: tests/v04/test_omero.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+from pathlib import Path
+
+from ome_zarr_models.v04.omero import Channel, Omero, Window
+
+
+def test_load_example_json():
+    with open(Path(__file__).parent / "data" / "omero_example.json") as f:
+        data = json.load(f)
+
+    print(Omero(**data))
```
Comment (d-v-b) [inline tests/v04/test_omero.py:11]: do we need to print here?

File: tests/v04/test_omero.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import json
+from pathlib import Path
+
+from ome_zarr_models.v04.omero import Channel, Omero, Window
+
+
+def test_load_example_json():
+    with open(Path(__file__).parent / "data" / "omero_example.json") as f:
```
Comment (d-v-b) [inline tests/v04/test_omero.py:8]: we should put these JSON documents in our tests module

File: tests/v04/test_omero.py (hunk 1 above)
Comment (dstansby) [reply tests/v04/test_omero.py:11]: lol woops

File: tests/v04/test_omero.py (hunk 2 above)
Comment (dstansby) [reply tests/v04/test_omero.py:8]: Isn't this the test module? What filepath within the repo are you rpoosing instead?

File: tests/v04/test_omero.py (hunk 1 above)
Comment (dstansby) [reply tests/v04/test_omero.py:11]: This is why we do review 😆

File: tests/v04/test_omero.py (hunk 2 above)
Comment (d-v-b) [reply tests/v04/test_omero.py:8]: quite right, `Path(__file__).parent / "data" / "omero_example.json"` will resolve to a path inside the tests module, but we could also store the JSON inline as python strings, and then they can be imported without requiring tests to do IO

Comment (d-v-b) [reply tests/v04/test_omero.py:8]: (another option would be keep the JSON files as JSON files, but make our tests access them via a pytest fixture that does the IO once per test session)

Comment (dstansby) [reply tests/v04/test_omero.py:8]: The advantage of storing in a .json file instead of an inline string is getting nice linting/syntax highlighting for it. At the moment this is only read in once per session - perhaps we can leave as is for now, and if/when we add more similar tests think about how to de-duplicate the machinery?

Comment (d-v-b) [reply tests/v04/test_omero.py:8]: +1 to storing JSON as JSON :) and this is not a blocker, we can handle this when we apply a good first-pass normalization of all our tests

File: tests/v04/test_omero.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                window=Window(max=65535.0, min=0.0, start=0.0, end=1500.0),
+                active=True,
+                coefficient=1,
+                family="linear",
+                inverted=False,
+                label="LaminB1",
+            )
+        ],
+        id=1,
+        name="example.tif",
```
Comment (d-v-b) [inline tests/v04/test_omero.py:24]: haha

**Comment by d-v-b** [comment]
re: the extra fields, that's a good and essential addition. I know of at least one OME-Zarr implementation that uses extra metadata fields (`ngff-zarr`)

**Review by d-v-b — APPROVED** [review]
lgtm!
